File: backend/Application/Film/DTOs/FilmDto.cs

namespace Application.Film.DTOs
{
    public class FilmDto
    {
        public int Id { get; set; }
        public Guid CreatorId { get; set; }
        public string Name { get; set; } = string.Empty;
        public int YearOfRelease { get; set; }
        public int Duration { get; set; }
        public string Description { get; set; } = string.Empty;
        public string LinkToPoster { get; set; } = string.Empty;
        public string LinkToFilm { get; set; } = string.Empty;
        public string Country { get; set; } = string.Empty;
        public double Rating { get; set; }
        public int AgeRestriction { get; set; }
        public List<int> GenreIds { get; set; } = new();
        public List<FilmPersonDto> Persons { get; set; } = new();


    }
}

File: frontend/src/types/FilmDto.ts
export type FilmPersonDto = {
  personId: number;
  careerId: number;
};

export type FilmDto = {
  id: number;
  creatorId: string;
  name: string;
  yearOfRelease: number;
  duration: number;
  description: string;
  linkToPoster: string;
  linkToFilm: string;
  country: string;
  rating: number;
  ageRestriction: number;
  genreIds: number[];
  persons: FilmPersonDto[];
};

export type CreateFilmDto = {
  creatorId: string;
  name: string;
  yearOfRelease: number;
  duration: number;
  description: string;
  linkToPoster: string;
  linkToFilm: string;
  country: string;
  ageRestriction: number;
  genreIds: number[];
};

export type UpdateFilmDto = Omit<CreateFilmDto, 'creatorId'>;
